injector/network_disruption.go

```diff
@@ -552,6 +552,28 @@ func (i *networkDisruptionInjector) buildServiceFiltersFromService(service v1.Se
 	return endpointsToWatch
 }
 
+func (i *networkDisruptionInjector) extractValidPortListInServicePorts(serviceSpec v1beta1.NetworkDisruptionServiceSpec, k8sService *v1.Service) []v1.ServicePort {
+	ports := k8sService.Spec.Ports
```

Comment:
WDYT of providing the intent this way:
* if `len(serviceSpec.Ports) == 0` then `return k8sService.Spec.Ports`
  * early quit, our most common case for now I think

Then, you could build a dictionary of "allowedPorts" from `serviceSpec.ports` with a key similar to:
* `<name|port>-<protocol(default=tcp)>`

Then, you compare `k8sService.Spec.Ports` to such dictionary and add it to `ports` if matched.